File: 495.py
class Solution:
    def findPoisonedDuration(self, timeSeries: List[int], duration: int) -> int:
        timeCount = 0
        currTime = 0
        endTime = 0
        for t in timeSeries:
            timeCount += min(endTime, t) - currTime
            endTime = t + duration
            currTime = t
        return timeCount + duration

File: sol/495.cpp
struct Solution {
    inline int findPoisonedDuration(vector<int>& timeSeries, int duration) {
        int count = 0, curr = 0, end = 0;
        for (int t : timeSeries){
            count += min(end, t) - curr;
            curr = t;
            end = t + duration;
        }
        return count + duration;
    }
};
